src/Compilers/Core/Portable/CommandLine/AnalyzerConfigSet.cs

```diff
@@ -94,12 +94,19 @@ public bool Equals([AllowNull] List<Section> x, [AllowNull] List<Section> y)
                 isEnabledByDefault: true);
 
         public static AnalyzerConfigSet Create<TList>(TList analyzerConfigs) where TList : IReadOnlyCollection<AnalyzerConfig>
+        {
+            return Create(analyzerConfigs, out _);
+        }
+
+        public static AnalyzerConfigSet Create<TList>(TList analyzerConfigs, out ImmutableArray<AnalyzerUnsetKey> unsetGlobalKeys) where TList : IReadOnlyCollection<AnalyzerConfig>
         {
             var sortedAnalyzerConfigs = ArrayBuilder<AnalyzerConfig>.GetInstance(analyzerConfigs.Count);
             sortedAnalyzerConfigs.AddRange(analyzerConfigs);
             sortedAnalyzerConfigs.Sort(AnalyzerConfig.DirectoryLengthComparer);
 
-            return new AnalyzerConfigSet(sortedAnalyzerConfigs.ToImmutableAndFree());
+            var mergedConfigs = GlobalAnalyzerConfigBuilder.FilterGlobalConfigs(sortedAnalyzerConfigs.ToImmutableAndFree(), out unsetGlobalKeys);
```

Comment:
The implementation of `FilterGlobalConfigs` treats the first parameter essentially as an `IEnumerable`. Not sure it's valuable to pay the cost of `ToImmutable` here.